assets/js/custom/ProgramCredits.js

```diff
@@ -1,50 +1,191 @@
 import $ from 'jquery'
+import LanguageApi from '../api/LanguageApi'
 /* global Routing */
 
-export function ProgramCredits (programId) {
+export function ProgramCredits (programId, translationSaved, translationDeleted, creditsSelect,
+  closeEditorDialog, keepOrDiscardDialog, snackbarNotification, customTranslationApi) {
   const credits = $('#credits')
   const editCreditsUI = $('#edit-credits-ui')
   const editCredits = $('#edit-credits')
+  const editCreditsButton = $('#edit-credits-button')
+  const closeCreditsEditorButton = $('#close-credits-editor-button')
+  const creditLanguageSelectorList = $('#credit-language-selector-list')
+  const creditsSelectedText = $('#credits-selected-text')
+  const editCreditsSubmitButton = $('#edit-credits-submit-button')
   const editCreditsError = $('#edit-credits-error')
   const descriptionCreditsContainer = $('#description-credits-container')
   const showMoreToggle = $('#descriptionShowMoreToggle')
   const descriptionHeadline = $('#description-headline')
 
-  $('.edit-credits-button').on('click', () => {
-    if (editCreditsUI.hasClass('d-none')) {
-      descriptionCreditsContainer.hide()
-      descriptionHeadline.hide()
-      editCreditsUI.removeClass('d-none')
-      showMoreToggle.addClass('d-none')
-    } else {
-      descriptionCreditsContainer.show()
-      descriptionHeadline.show()
-      editCreditsUI.addClass('d-none')
-      handleShowMore()
-    }
+  let languages = {}
+  let previousCreditsIndex = 0
+  let lastSavedCredits = credits.text().trim()
+
+  $(document).ready(function () {
+    LanguageApi.getLanguages(populateSelector)
   })
 
-  $('#edit-credits-submit-button').on('click', () => {
+  function populateSelector (languageList) {
+    languages = languageList
+    creditLanguageSelectorList.empty()
+    creditLanguageSelectorList.append('<li class="mdc-list-item" data-value="default" role="option" tabindex="-1">' +
+        '<span class="mdc-list-item__ripple"></span>' +
+        '<span class="mdc-list-item__text">Default</span>' +
+        '</li>')
+
+    for (const language in languages) {
+      if (language.length <= 2) {
+        creditLanguageSelectorList.append(`<li class="mdc-list-item" data-value="${language}" role="option" tabindex="-1">\
+          <span class="mdc-list-item__ripple"></span>\
+          <span class="mdc-list-item__text">${languages[language]}</span>\
+          </li>`)
+      }
+    }
+
+    creditsSelect.layoutOptions()
+    resetCreditsEditor()
+  }
+
+  function getCustomTransalationSuccess (data) {
+    editCredits.val(data)
+    lastSavedCredits = data
+    editCredits.removeAttr('disabled')
+  }
+
+  function getCustomTransalationError () {
+    const defaultCredits = credits.text().trim()
```

Comment:
i think it's better to show empty string when the api returns 404 when no translation is defined

Reply:
Updated to show an empty text field when there wasn't a translation defined